wgpu-hal/src/vulkan/device.rs

```diff
@@ -1021,6 +1021,93 @@ impl super::Device {
     pub fn shared_instance(&self) -> &super::InstanceShared {
         &self.shared.instance
     }
+
+    fn check_for_oom(&self, needs_host_access: bool, size: u64) -> Result<(), crate::DeviceError> {
```

Comment:
I don't get the difference between check_for_oom and check_if_oom - they definitely need more descriptive names

Reply:
I agree the names are not great, `check_if_oom` checks if we are currently out of memory, `check_for_oom` checks if we will be out of memory if we were to allocate the given resource.

I'm thinking to rename:

- `check_if_oom` to `error_if_oom`
- `check_for_oom` to `error_if_oom_on_resource_allocation`

Thoughts?

Reply:
What about:

- `error_if_oom`
- `error_if_would_oom_on_resource_allocation`

I don't love amazingly long names but can't think of anything better